src/api_server/src/lib.rs

```diff
@@ -129,6 +136,12 @@ impl ApiServer {
         loop {
             match server.requests() {
                 Ok(request_vec) => {
+                    if let Some(exit_code) = *self.encountered_fatal_error.borrow() {
+                        unsafe {
+                            error!("Fatal error with exit code: {}", exit_code);
```

Comment:
nit: take this `error!` out of the `unsafe` block.

Reply:
Will do.